apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/SpansResourceTest.java

```diff
@@ -2766,7 +2768,79 @@ void getByProjectName__whenFilterFeedbackScoresLessThanEqual__thenReturnSpansFil
             getAndAssertPage(workspaceName, projectName, filters, spans, expectedSpans, unexpectedSpans, apiKey);
         }
 
-        static Stream<Filter> getByProjectName__whenFilterInvalidOperatorForFieldType__thenReturn400() {
+        Stream<Arguments> getSpansByProject__whenFilterByDuration__thenReturnSpansFiltered() {
+            return Stream.of(
+                    arguments(Operator.EQUAL, Instant.now().truncatedTo(ChronoUnit.MILLIS),
+                            Duration.ofMillis(1L).toNanos() / 1000, 1.0),
+                    arguments(Operator.GREATER_THAN, Instant.now().truncatedTo(ChronoUnit.MILLIS),
+                            Duration.ofMillis(8L).toNanos() / 1000, 7.0),
+                    arguments(Operator.GREATER_THAN_EQUAL, Instant.now().truncatedTo(ChronoUnit.MILLIS),
+                            Duration.ofMillis(1L).toNanos() / 1000, 1.0),
+                    arguments(Operator.GREATER_THAN_EQUAL, Instant.now().truncatedTo(ChronoUnit.MILLIS),
+                            Duration.ofMillis(1L).plusNanos(1000).toNanos() / 1000, 1.0),
+                    arguments(Operator.LESS_THAN, Instant.now().truncatedTo(ChronoUnit.MILLIS),
+                            Duration.ofMillis(1L).plusNanos(1).toNanos() / 1000, 2.0),
+                    arguments(Operator.LESS_THAN_EQUAL, Instant.now().truncatedTo(ChronoUnit.MILLIS),
+                            Duration.ofMillis(1L).toNanos() / 1000, 1.0),
+                    arguments(Operator.LESS_THAN_EQUAL, Instant.now().truncatedTo(ChronoUnit.MILLIS),
+                            Duration.ofMillis(1L).toNanos() / 1000, 2.0));
+        }
+
+        @ParameterizedTest
+        @MethodSource
+        void getSpansByProject__whenFilterByDuration__thenReturnSpansFiltered(Operator operator, Instant start,
+                long end, double duration) {
```

Comment:
Why do we need to have `Instant start` as parameter? It's always the same and can be just instantiated within the test.
And for the end you could just pass span duration as long (1L for example) and do conversion in the test: `Duration.ofMillis(1L).plusNanos(1).toNanos() / 1000`. Will make arguments much shorter and more readable.

Reply:
Good point!